Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -127,12 +127,16 @@ private extension PointOfSaleCouponService {
     }
 
     func mapCouponsToPOSItems(coupons: [Coupon]) -> [POSItem] {
-        coupons.compactMap { coupon in
-                .coupon(POSCoupon(
-                    id: UUID(),
-                    code: coupon.code,
-                    summary: coupon.summary(currencySettings: currencySettings)
-                ))
+        let now = Date()
+        let nonExpiredCoupons = coupons.filter { coupon in
+            guard let expirationDate = coupon.dateExpires else { return true }
+            return expirationDate >= now
+        }
+
+        return nonExpiredCoupons.compactMap { coupon in
+                .coupon(POSCoupon(id: UUID(),
+                                  code: coupon.code,
+                                  summary: coupon.summary(currencySettings: currencySettings)))
```

Comment:
nit; you can chain .filter { } .compact { } functional programming way

```suggestion
        return coupons
            .filter {
                guard let expirationDate = $0.dateExpires else { return true }
                return expirationDate >= Date()
            }
            .compactMap {
                .coupon(
                    POSCoupon(
                        id: UUID(),
                        code: $0.code,
                        summary: $0.summary(currencySettings: currencySettings)
                    )
                )
            }
```

Reply:
TIL! I'll leave it with the existing implementation for now just because I feel it's more explicit, which makes it easier to read.